packages/promotion/src/repositories/campaign.ts

```diff
@@ -8,4 +10,60 @@ export class CampaignRepository extends DALUtils.mikroOrmBaseRepositoryFactory<
     create: CreateCampaignDTO
     update: UpdateCampaignDTO
   }
->(Campaign) {}
+>(Campaign) {
+  async create(
```

Comment:
question: Shouldn't this logic live at the service level as per our convention? Or is this a special case?

Reply:
This is to add promotions to the campaign collection (through `.add()`), which is purely dependent on MikroORM. If we add it to the service, we make the service mikroorm dependent.

Reply:
Yes, I think we're aligned on that. But it is true that ORMs handle one to many or many to many relationships differently. Shouldn't we just have to ensure that we pass a consistent shape to the repository layer and then the repositories handle them based on how they process relationships?